### PR TITLE
[Experimental] Show logs by task

### PR DESCRIPTION
**Pull request by serihiro**
# What is this PR ?
- This PR is POC implementation of digdag-ui
- This PR modifies the format of task log in views so that logs are splitted for each tasks.

# Background of this PR
- Currently all task logs are unified, so it is difficult to distinguish that what task each line is  belongs to.

# e.g.

screen shot 2018-08-27 at 12 21 05

### CODE REVIEW

**Comment by muga** [comment]
@serihiro I like this idea that you're suggesting. Because I actually faced similar problem that it was not easy to find a specific task log messages in the unified log messages. it would be helpful for my case.

This change looks good to me basically. This is not trivial but, if we could introduce the link to each task log in the task in "Tasks" section, it would be great. Users first check task failures in "Tasks" section and may want to see those task logs on "Logs" section. We'd better to introduce association or something between "Tasks" and "Logs" sections I think. Please take a look if you get a chance.

**Comment by serihiro** [comment]
@muga 
I added hash link like this. ( c9e0a3f2de029b55e809666bf2b121ab255686e8 )
When user clicks a taskId then jump to the LogFileView of the clicked taskId. How about this?

digdag-ui-

**Comment by hiroyuki-sato** [comment]
Hello, @serihiro 

This is a comment from a user. 

This change seems totally good. 
My favorite style is a collapse and expands view.
If a user clicks taskId link, It expands ( or collapse) the log view.

**Comment by serihiro** [comment]
Hello @hiroyuki-sato . 

Thank you for your comment !

Your suggestion sounds good. But, we think that there is few differences between current implementation and your idea, in a point of view achieving the purpose(being easy to read each task log).

So, experimentally we plan to introduce the current implementation. Of course, good opinions are always welcome, and we can keep improving digdag-ui. If you have more good idea, let's discuss with issue or PR.

**Comment by muga** [comment]
@hiroyuki-sato If you could make sharing the comment from the user who we are missing, it would be great to improve UI. 

@serihiro LGTM. Please let me know if you still have any change and improvement that you're expecting before merging this PR into master?

**Comment by serihiro** [comment]
@muga 
Thank you for your review! I do not have any remaining tasks about this PR.

**Comment by muga** [comment]
@dzannotti It would be great if you can take a look this PR.

File: digdag-ui/console.jsx [hunk 1]
```diff
@@ -1763,9 +1774,17 @@ class LogFileView extends React.Component {
...
 
   render () {
-    return this.state.data
-      ? <span>{pako.inflate(this.state.data, {to: 'string'})}</span>
-      : null
+    if(this.state.data){
+      return (
+        <div>
+          <h3 id={'logs-' + this.props.file.taskName + this.props.order.toString()}
+              style={{display: 'block', paddingTop: '50px', marginTop: '-50px'}}>{this.props.file.taskName}</h3>
```
Comment (dzannotti) [inline digdag-ui/console.jsx]: this shouldn't be an inline `style` should be part of style.less

File: digdag-ui/console.jsx [hunk 2]
```diff
@@ -1763,9 +1774,17 @@ class LogFileView extends React.Component {
...
+          <pre>{pako.inflate(this.state.data, {to: 'string'})}</pre>
+        </div>
+      )
+    }else{
```
Comment (dzannotti) [inline digdag-ui/console.jsx]: nit: this doesn't pass linting rules

File: digdag-ui/console.jsx [hunk 3]
```diff
@@ -1796,6 +1815,23 @@ class AttemptLogsView extends React.Component {
     this.fetch()
   }
 
+  logFiles () {
+    if (!this.state.files.length) {
```
Comment (dzannotti) [inline digdag-ui/console.jsx]: I think this would read better if destructured `const { files } = this.state`

**Comment by muga** [comment]
@dzannotti thank you for reviewing! let you know again once fixing them with @serihiro

**Comment by serihiro** [comment]
@dzannotti 
Thank you for your reviewing and comments!
I fixed by your comments. Could you check again ?

**Comment by dzannotti** [comment]
👍 LGTM

**Comment by serihiro** [comment]
@dzannotti 
Thank you for your reviewing !

@muga 
I fixed all issues. Could you take care this PR ?

**Comment by muga** [comment]
@serihiro @dzannotti Thank you! I will merge once CI will pass.

**Pull request merged**